aten/src/ATen/native/ForeachUtils.h

```diff
@@ -151,6 +163,13 @@ bool can_use_fast_route(TensorList tensors1, TensorList tensors2) {
     if (!has_same_attributes(expected_device, {tensors1[i], tensors2[i]})) {
       return false;
     }
+
+    // In case of division, integer inputs will result in float
+    if (division_op) {
+      if (at::isIntegralType(tensors1[i].scalar_type(), /*includeBool*/ true)) {
+        return false;
+      }
+    }
```

Comment:
It bothers me a little that we have so many `can_use_fast_route` overloads that do roughly the same thing. I wonder if there's a way to refactor the code so that we have even fewer of them

Also, does `bool can_use_fast_route(TensorList tensors1, TensorList tensors2, Scalar scalar)` also need to be changed?

Reply:
refactored. please, let me know if i should break `can_use_fast_route` refactoring into a separate PR.

Reply:
Nah, this is fine